Buddies/Utilities/FirestoreManager.swift

```diff
@@ -36,40 +36,42 @@ class FirestoreManager {
     }
     
     static func getUserAssociatedActivities( userID: String,
-                                             _ callback: @escaping ([[ActivityId]]) -> ()) {
+                                             _ callback: @escaping ([[Activity]]) -> ()) {
         Firestore.firestore().collection("activities").whereField("members", arrayContains: userID).getDocuments { result, error in
             guard let result = result,
                 error == nil else {
                     print("Error loading \(userID) from Firestore: \n \(String(describing: error))")
-                    callback([])
+                    callback([[],[],[]])
                     return
             }
             
-            var created = [ActivityId]()
-            var joined = [ActivityId]()
-            var previous = [ActivityId]()
+            var created = [Activity]()
```

Comment:
Curious about this change to "Activity" instead of IDs. Don't we only need the ID in class bc we need to pass it into useActivities?

Reply:
It's a bit of a hack to solve a chicken and egg situation resulting from designing expecting a third party to handle filtering (i.e. algolia).  We need the activities' to be loaded (with description/title/etc) to decide which activities should be loaded (match the search).  We need to know before we load them so we can bind a callback that calls `reloadAtIndex` for each activity.

We can't know the index of an activity until we know which preceding activities match the filter, and that's all async.  The hack is to use Firebase to get the params on which we filter, then use the data accessor to load (with callbacks and updates set up) the activities that match.  

The parent `ActivityTableVC` was built around an assumption that, at load time, the activity IDs were sorted correctly. If we don't care about sorting (which we do), we could override `loadActivity` to ignore the initial ordering and store activities in the order they are loaded.

Reply:
Interesting stuff! I trust you ;)